### ISSUE TITLE
Bug: [await-thenable] fails to pick up on a ternary's condition not being a Promise

### ISSUE THREAD

**Issue by ajvincent** [typescript-eslint/typescript-eslint#6283]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.9.4&sourceType=module&showAST=ts&code=IYZwngdgxgBAZgV2gFwJYHsL3egFADwC4YRkAnVCAcwEoZiAFM9AW1RAFMAeAN3VQAmAPhgBvAFAwYwAO7BUyGPhgBeNTABEACyoaYAfklSYTVuw4A6MhxDoANjw64AjHUJGpptpys37j3AAmGg8YAG5xAF9xPkFcOBwaCKA&eslintrc=N4KABGBECmB2BukBcZQQjAzgJgAzYEZkwAXAJwFdoAacdSWAewBNpjyq6BfWjaADxJxmmYgG06fTABsAlrBJIy0AMaMAtuuHRmkXvQAO0igHN5SAAIkAngeiYVZWQZIBae3IUB6D-LfK1TW1dfQwjU3MrW3tHZzdfbwCNLVhWEMkocLNYSxs7BycXdxk-LySg1J1XZQBHClknWBNXPOhXFQALVQBreRNIOgBdfUgDAEMyTGgyYkgo-NiihJIvccnpvTpRiamyAHkXWUZYURQ0ehIHY4AzWRMAJUZGEgARBtmAOk30TLJGACtVCRxBkMB8vJc1LBbiYPv9MMcBj9hhkYCp1GMAGrTTBHWCzaRjISYYGhKAIihkFTQAAq0Vm6hYFGkbG4Iyy8lOYAkPzmrQKcWKnlJqOWrg5sBamGYjBUSLAKIwlBZXPOGHmMUK8RK3gERlkKlkbmuFFgKhIeOq0BIlMlrRBPz4ZD+Mwyg24IC4QA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkqyARhWANrg23okBMADN90gAadh0gATBJFFgAuiLHxY4gK4ZmkLnwHCZkPAENMAc3R4NW-oIU1IS1RgBKXWKhUFiGosvS6xJWBVMaHQAWQNEZjxMFXQbWnF0aFQjAw8iKJi49gBfG00ADzx0InEySkgAATwSOCIAM2RjAHpvRIBGAA5mmrrG4wA6ACsAjJAcoA

### Repro Code

```typescript
async function foo(x: string) : Promise<void> {
  await x === "hg" ?
    Promise.resolve(1) :
    Promise.resolve(2)
    ;
}
void(foo);
```

### ESLint Config

```javascript
{
  "env": {
    "es2021": true,
    "node": true
  },
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/eslint-recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:@typescript-eslint/recommended-requiring-type-checking"
  ],
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "tsconfigRootDir": ".",
    "project": ["./tsconfig.json"],
    "ecmaVersion": "latest",
    "sourceType": "module"
  },
  "plugins": [
    "@typescript-eslint"
  ]
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "lib": ["es2022", "dom"],
    "module": "es2022",
    "target": "es2022",
    "moduleResolution": "node"
  },

  "extends": "@tsconfig/node18/tsconfig.json"
}
```

### Expected Result

I expected the second line (`await x === "hg"`) would trigger a warning.

### Actual Result

There was no warning for the condition of the ternary not being a thenable.

### Additional Info

This bit me as I'd triggered async operations which I wasn't awaiting as I thought I was, including file-system operations.  It led me down a rabbit hole, thinking NodeJS was exiting a function incorrectly, when the async function to run in the truthy case had an await statement of its own...

**Comment by ajvincent** [comment]
Invalid.  When I look at it in Visual Studio Code with my updated linter configuration, it shows up bright red.  Sorry for the spam!
